components/sidebar.jsx

```diff
@@ -5,27 +5,85 @@ var LinkAnchorSwap = require('./link-anchor-swap.jsx');
 
 var Login = require('./login.jsx');
 
+var Subitems = React.createClass({
+  render: function() {
+    var items = this.props.subItems.map(function (item, key) {
+      return (
+        <li key={item.name}>
+          <Link to={item.link}>
+            {item.name}
+          </Link>
+        </li>
+      )}
+    );
+    return (
+      <ul className="sidebar-subitems">
+        {items}
+      </ul>
+    );
+  }
+});
+
+var TopLevelNavItem = React.createClass({
+  getInitialState: function() {
+    return {
+      activeSubNav: false
+    };
+  },
+  componentDidMount: function() {
+    this.highlightActiveSubNav();
+  },
+  componentDidUpdate: function() {
+    this.highlightActiveSubNav();
+  },
+  highlightActiveSubNav: function() {
+    var activeState = !!this.getDOMNode().querySelector(".sidebar-subitems a.active");
```

Comment:
the fact that we need DOM querying here is a sign we're doing something wrong. React doesn't need DOM queries for anything, if done right; this should be a thing we can simply ask the relevant `this.refs` component. What does this code achieve, and can we do that "directly"?